src/python/pants/backend/python/typecheck/mypy/subsystem_test.py

```diff
@@ -196,8 +196,8 @@ def assert_lockfile_request(
         MyPy.default_interpreter_constraints,
     )
 
-    # If no Python targets in repo, fall back to MyPy constraints.
-    assert_lockfile_request("target()", MyPy.default_interpreter_constraints)
```

Comment:
I don't quite understand this change. But if there are no python targets in the repository, then it probably doesn't matter which we use...?

Reply:
It's because the rule helper says `or python_setup.interpreter_constraints` when there were no interpreter constraints. Whereas the old bespoke code said `or mypy.interpretre_constraints`. That's pretty unlikely to happen -- it's only when there are zero MyPy compatible targets in the repo.